### PR TITLE
Add support for Babel

### PR DESCRIPTION
**Pull request by jamiebuilds**

### CODE REVIEW

**Comment by silvenon** [comment]
Hey, thanks for the awesome PR :+1: The team decided to completely replace CoffeeScript with Babel, would you care to add that to your PR too?

**Comment by jamiebuilds** [comment]
Will do

**Comment by jamiebuilds** [comment]
This might be pushing it, but I could make the change that Babel is always included. If people do not want to use ES6 features, Babel would just pass the file right through without transforming anything. It would simplify the repo a bit more and encourage people to start using ES6+.

**Comment by arthurvr** [comment]
Nah. I don't really know if all-day users would like that. Generally I don't like it if my build system is cluttered with things I don't use (in case they wouldn't use ES6), so I don't think they will eighter :/

**Comment by jamiebuilds** [comment]
Alright, removed CoffeeScript

File: app/templates/Gruntfile.js [hunk 1]
```diff
@@ -405,17 +405,17 @@ module.exports = function (grunt) {
...
-      server: [<% if (coffee) {  %>
-        'coffee:dist',<% } %><% if (includeSass) { %>
+      server: [<% if (babel) {  %>
+        'babel:dist',<% } %><% if (includeSass) { %>
         'sass:server'<% } else { %>
         'copy:styles'<% } %>
       ],
-      test: [<% if (coffee) { %>
-        'coffee'<% } %><% if (coffee && !includeSass) {  %>,<% } %><% if (!includeSass) { %>
+      test: [<% if (babel) {  %>
```
Comment (arthurvr) [inline app/templates/Gruntfile.js]: Why did you add that space?

File: app/templates/Gruntfile.js [hunk 2]
```diff
@@ -405,17 +405,17 @@ module.exports = function (grunt) {
...
+        'babel'<% } %><% if (babel && !includeSass) {  %>,<% } %><% if (!includeSass) { %>
         'copy:styles'<% } %>
       ],
-      dist: [<% if (coffee) { %>
-        'coffee',<% } %><% if (includeSass) { %>
+      dist: [<% if (babel) {  %>
```
Comment (arthurvr) [inline app/templates/Gruntfile.js]: Let's remove that extra space :)

**Comment by stevemao** [comment]
Any reason why you guys want to remove CoffeeScript? Does it apply to other generators?

**Comment by silvenon** [comment]
CoffeeScript was created to patch some shortcomings of JavaScript (other than to provide some syntactic sugar). Now that we have an [ES6 transpiler](https://babeljs.io/docs/learn-es6/), there's no real need to support CoffeeScript anymore. We generally advise using plain JS over CoffeeScript because it's native (i.e. new features will be faster), the barrier of entry is lower (e.g. Atom being written in CS [increases the barrier](https://twitter.com/sindresorhus/status/591869731109343233)) and, in my opinion, there is less and less reason for it to exist.

Yes, we'll probably drop it from other generators as well, eventually.

File: test/test.js [hunk 3]
```diff
@@ -80,26 +78,6 @@ describe('Webapp generator', function () {
...
-
-        assert.fileContent([].concat(
-          expectedContent,
-          [['Gruntfile.js', /coffee/]]
-        ));
-
-        done();
-      });
-    });
-
```
Comment (arthurvr) [inline test/test.js:102]: Should probably test the babel option in some way

Comment (arthurvr) [inline test/test.js:102]: ping? I just think that we should at least have a test for the `babel` option.

Comment (jamiebuilds) [inline test/test.js:102]: added

**Comment by arthurvr** [comment]
I like this :+1: @silvenon WDYT? Good to go?

**Comment by addyosmani** [comment]
LGTM for a merge. 

Side: It's pretty great to see the Babel community proactively helping existing tools improve their ES6 support. Thanks a lot @thejameskyle!

**Comment by addyosmani** [comment]
@silvenon In agreement regarding our general dropping of CS. Now might be a good time for us to open issues for any generators with remaining CS support to consider switching those over to ES6 too.

**Comment by silvenon** [comment]
@addyosmani good idea. Done.

**Comment by silvenon** [comment]
I tried it, seems to work fine. Very good job, @thejameskyle, thanks a lot :+1: 

I would even go as far as not making Babel optional, because it's just JavaScript, but this will do perfectly for now.

**Comment by addyosmani** [comment]
:+1:

**Comment by jamiebuilds** [comment]
Thanks guys!

**Comment by arthurvr** [comment]
> Side: It's pretty great to see the Babel community proactively helping existing tools improve their ES6 support

Sure! :+1: Thanks @thejameskyle!

> In agreement regarding our general dropping of CS. Now might be a good time for us to open issues for any generators with remaining CS support to consider switching those over to ES6 too.

If that's the plan maybe we should consider writing about it at some point too? (a blogpost I mean) It's a change that significantly impacts our users and it's all about shiny new technology. There's much interesting stuff going on in the world of ES2015-tooling these days.

**Comment by jamiebuilds** [comment]
You might also want to switch to eslint with [babel-eslint](https://github.com/babel/babel-eslint) as the parser for better ES6/ES7/JSX/etc support

**Comment by silvenon** [comment]
Thanks for the tip! I've just seen it recently, will check it out :+1:
